### ISSUE TITLE
Option to disable the 3D view

### ISSUE THREAD

**Issue by TomJansen** [westnordost/StreetComplete#966]
It would be nice if there was an option to the disable the 3D view and only have the eagle view (2D, directly from above). Right now, the app defaults to the 3D view and I have to swipe with two fingers to enable 2D view.

Thank you for this app.

**Comment by rugk** [comment]
Actually, you can just tap on the compass and it should disable 3d.
I don't know how one would do that with a two finger swipe.

**Comment by TomJansen** [comment]
>I don't know how one would do that with a two finger swipe.
swipe up or down with 2 fingers

**Comment by ENT8R** [comment]
I think having such an option in the settings is too specific and kind of contraproductive to the simplicity of the app... Nowadays nearly all applications with a map have the option to display a 3D map, so this is not something very special...

**Comment by TomJansen** [comment]
> Nowadays nearly all applications with a map have the option to display a 3D map, so this is not something very special

It is true that almost every map has this, but with 3D view, one loses the overview and clarity that is needed with Streetcomplete

**Comment by ENT8R** [comment]
But what speaks then against tapping on the compass to bring everything back to the normal view?

**Comment by matkoniecz** [comment]
Blocked by #673

**Comment by rugk** [comment]
This has nothing to do with the mapstyle.

**Comment by matkoniecz** [comment]
Current mapstyle has 3D buildings in every map mode. As I understand this issue @TomJansen wants to disable this feature - and it requires ability to select one of available maps styles.

**Comment by ENT8R** [comment]
But the other mapstyles (except the satellite style) will have 3D buildings too and you can rotate the map using two fingers swiping up and down even in the satellite style...

**Comment by matkoniecz** [comment]
That is why it is a separate issue, as it requires

- adding ability to change map styles

- adding map style without 3D buildings

**Comment by ENT8R** [comment]
First, 3D buildings were wanted (#183) and now we should remove them again? But why? I don't see a reason for that...

**Comment by matkoniecz** [comment]
I guess that some people want them and some prefer to not have them.

I am not sure what is preference of @westnordost - one style (rejecting #673 and this)? Multiple selectable styles? Who would maintain list of styles?

**Comment by goldfndr** [comment]
@matkoniecz I'm pretty sure that this issue is for having the tangram [Tilt](https://mapzen.com/documentation/tangram/android-sdk/0.9.0/com/mapzen/tangram/MapController.html#setTilt-float-) always be overhead (0), not for buildings to be rendered without height.

@rugk Regardless of compass glued or not, tilting is available for me, and the value doesn't change when I glue/unglue.

@TomJansen Are you accidentally tilting when you manually rotate or zoom? Or is the tilting occurring elsehow? For me, the last tilt value is kept each time I launch StreetComplete.

**Comment by forght** [comment]
I just found the tilt function through this post, I always just used the top down fixed north view until now and that ist default at startup.
As a see it now there are two possible startup modes:
- top down fixed north
or
- tilted fixed to compass
and switching from one two the other forces the camera to tilt.

Wouldn't it solve @TomJansen issue to save the default tilt independent from the chosen compass view and not force the tilt when switching?
In this way He could once tilt to top down view and switch between compass and north fixed view without getting forced to the tilted view every time.

**Comment by westnordost** [comment]
I am not convinced that this option is necessary. If users feel that 3d buildings are obstrusive, then they can just look at the map from top down.

**Comment by TomJansen** [comment]
I like to cycle and then I often fill in quests with Streetcomplete. Therefore the top-view is handy. With tapping the compass, the view is fixed to norths, which is not really handy while cycling. I can have the top view with the direction folowing thing by swiping with 2 fingers down, but this is nearly impossible to do while cycling.

**Comment by ENT8R** [comment]
Did you try to tap on the location request button in the lower right corner and then tapping on the compass icon? Shouldn't this then automatically let the map rotate while the view is a 2D view?

**Comment by ENT8R** [comment]
Alright. Nevermind my last comment. I just tested it and what I described does only enable the 3D view. Maybe this issue can be fixed by introducing a third state to the compass icon?

**Comment by exploids** [comment]
I personally find the 3D buildings really annoying and wish there was just a simple switch to disable them. I already only use the top down view, but that doesn't change the fact that the buildings are still 3D.

**Comment by mnalis** [comment]
see also https://github.com/streetcomplete/StreetComplete/issues/2206

**Comment by StyXman** [comment]
> tap on the compass and it should disable 3d.

Is it me or if I don't have the GPS, or don't rotate the map, I don't get the compass? I'll post a screenshot soon, but the icon is not present. If I rotate the map, one prees on the icon puts it North up and dissapears.

**Comment by smichel17** [comment]
- That comment is 4 years old and the controls have changed since then.
- It it is still true that tapping the compass will tilt the map so it is flat (and rotate so north is up). If the compass does not appear, the map is already flat.
- The app has been "birdseye 3d" when flat all along. However, 4 years ago we were still using the nextzen map tiles, which were even more out of date and were often missing building height information, so it was often hard to tell the difference between birdseye 3d and flat.

**Comment by StyXman** [comment]
then how is flat mode enabled now?

**Comment by westnordost** [comment]
There never was a flat mode

**Comment by StyXman** [comment]
out of honest ignorance, since I'm a recent user, have you profiled the app? do you know what makes people complain several (but maybe not enough) times about it's battery consumption? don't you think making the phone work less, by not animating stuff and rendering what some people think as obtrusive 3D buildings, would make it consume less battery? 

I might as well be wrong, but have you even tried?

**Comment by mnalis** [comment]
> then how is flat mode enabled now?

_"birdseye 3D mode"_ is now enabled by clicking on the GPS button (button in the lower-right corner, just below `+` and `-` buttons, looks either like black or orange dot, or like a black or orange arrowhead). Click if  once or twice and you should see the screen tilting from birdseye mode to more tilted mode. 

Note however that high buildings will still show their 3D-ness even in birdseye mode. To avoid that you need to install above-mentioned fork (it can be installed in parallel with upstream StreetComplete, which is useful for comparing/testing too)

**Comment by westnordost** [comment]
I can say with confidence that whether the map is displayed with 3D buildings or flat buildings will literally have no effect on battery consumption.

However, if anyone wants to experiment with possible performance improvements with the map, deactivate the showing of the view direction (compass direction) in the code. Currently, I think the whole map is re-rendered every frame because the compass sensor will update ~every frame and thus the view direction graphic is rendered onto the map every frame. If this is not done, this could drastically reduce the number of reduce the number of OpenGL drawing done.
Using the profiler builtin in Android Studio here will not be helpful because the map renderer is native code. If someone wants to experiment with it, he should maybe just do a normal mapping trip for e.g. 1 hour and see how much % of the battery that eats with the view direction and without the view direction.
